examples/nlp/dialogue_state_tracking/dialogue_state_tracking_sgd.py

```diff
@@ -0,0 +1,456 @@
+)
+parser.add_argument(
+    "--no_fuzzy_match",
+    action="store_true",
+    help="Whether to use fuzzy string matching when comparing non-categorical slot values. Fuzz match should not be used when conducting multiwoz style evaluation.",
+)
+parser.add_argument(
+    "--dialogues_example_dir",
+    type=str,
+    default="dialogues_example_dir",
+    help="Directory where preprocessed DSTC8 dialogues are stored.",
+)
+parser.add_argument(
+    "--overwrite_dial_files", action="store_true", help="Whether to generate a new file saving the dialogue examples."
+)
+parser.add_argument("--no_shuffle", action="store_true", help="Whether to shuffle training data")
+parser.add_argument("--no_time_to_log_dir", action="store_true", help="whether to add time to work_dir or not")
+parser.add_argument(
+    "--eval_dataset", type=str, default="dev", choices=["dev", "test"], help="Dataset split for evaluation."
+)
+parser.add_argument(
+    "--save_epoch_freq",
+    default=1,
+    type=int,
+    help="Frequency of saving checkpoint '-1' - step checkpoint won't be saved",
+)
+parser.add_argument(
+    "--save_step_freq",
+    default=-1,
+    type=int,
+    help="Frequency of saving checkpoint '-1' - step checkpoint won't be saved",
+)
+
+parser.add_argument(
+    "--loss_log_freq", default=-1, type=int, help="Frequency of logging loss values, '-1' - at the end of the epoch",
+)
+
+parser.add_argument(
+    "--loss_reduction",
+    default='mean',
+    type=str,
+    help="specifies the reduction to apply to the final loss, choose 'mean' or 'sum'",
+)
+
+parser.add_argument(
+    "--eval_epoch_freq", default=1, type=int, help="Frequency of evaluation",
+)
+
+parser.add_argument(
+    "--num_workers",
+    default=2,
+    type=int,
+    help="Number of workers for data loading, -1 means set it automatically to the number of CPU cores",
+)
+
+parser.add_argument(
+    "--enable_pin_memory", action="store_true", help="Enables the pin_memory feature of Pytroch's DataLoader",
+)
+
+parser.add_argument(
+    "--state_tracker",
+    type=str,
+    default='baseline',
+    choices=['baseline', 'ret_sys_act'],
+    help="Specifies the state tracker mode",
+)
+parser.add_argument(
+    "--schema_emb_init",
+    type=str,
+    default='baseline',
+    choices=['baseline', 'random', 'last_layer_average'],
+    help="Specifies how schema embeddings are generated. Baseline uses ['CLS'] token",
+)
+parser.add_argument(
+    "--train_schema_emb", action="store_true", help="Specifies whether schema embeddings are trainables.",
+)
+parser.add_argument(
+    "--debug_mode", action="store_true", help="Enables debug mode with more info on data preprocessing and evaluation",
+)
+
+parser.add_argument(
+    "--checkpoints_to_keep", default=1, type=int, help="The number of last checkpoints to keep",
+)
+
+args = parser.parse_args()
+logging.info(args)
+
+if args.debug_mode:
+    logging.setLevel(10)
```

Comment:
Should we rather us "constants"?
https://docs.python.org/3/library/logging.html#levels

And btw. why only debug?